pkg/apis/application/v1alpha1/types.go

```diff
@@ -1096,6 +1096,8 @@ type ResourceDiff struct {
 	NormalizedLiveState string `json:"normalizedLiveState,omitempty" protobuf:"bytes,9,opt,name=normalizedLiveState"`
 	// PredictedLiveState contains JSON serialized resource state that is calculated based on normalized and target resource state
 	PredictedLiveState string `json:"predictedLiveState,omitempty" protobuf:"bytes,10,opt,name=predictedLiveState"`
+	ResourceVersion    string `json:"resourceVersion,omitempty" protobuf:"bytes,11,opt,name=resourceVersion"`
+	SyncStatus         string `json:"syncStatus,omitempty" protobuf:"bytes,12,opt,name=syncStatus"`
```

Comment:
SyncStatus is not used in the code. As I mention in comment earlier we need to store previous diff result in cached ResourceDiff. Please replace `SyncStatus` with `Modified bool` and use in `diffArrayCached`